src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -231,9 +227,23 @@ public CompletableFuture<Void> generateIsland(Island island, Schematics.Schemati
             setHome(island, schematicConfig);
             deleteIslandBlocks(island).join();
             IridiumSkyblock.getInstance().getSchematicManager().pasteSchematic(island, schematicConfig).join();
-            setIslandBiome(island, XBiome.matchXBiome(schematicConfig.overworld.biome));
-            setIslandBiome(island, XBiome.matchXBiome(schematicConfig.nether.biome));
-            setIslandBiome(island, XBiome.matchXBiome(schematicConfig.end.biome));
+
+            for(Map.Entry<World.Environment, Boolean> enabledWorld : IridiumSkyblock.getInstance().getConfiguration().enabledWorlds.entrySet()) {
+                if(!enabledWorld.getValue()) {
+                    continue;
+                }
+
+                switch (enabledWorld.getKey()) {
+                    case NETHER:
+                        setIslandBiome(island, XBiome.matchXBiome(schematicConfig.nether.biome));
+                        break;
+                    case THE_END:
+                        setIslandBiome(island, XBiome.matchXBiome(schematicConfig.end.biome));
+                        break;
+                    default:
+                        setIslandBiome(island, XBiome.matchXBiome(schematicConfig.overworld.biome));
+                    }
+            }
```

Comment:
Can this section not stay how it is? since we already handle if the world is enabled in setIslandBiome no?

I think the previous code is alot cleaner

Reply:
ill do some testing, i also found it a bit redundant but i wanted to make sure we got rid of the problem

this still doesn't help anything if the worlds are enabled/disabled during server runtime - this is a separate issue tho